frontend/src/js/entity-history/timeline/ConceptName.tsx

```diff
@@ -0,0 +1,55 @@
+import styled from "@emotion/styled";
+import { memo } from "react";
+
+import { ColumnDescription, ConceptIdT, DatasetT } from "../../api/types";
+import { getConceptById } from "../../concept-trees/globalTreeStoreHelper";
+import { EntityEvent } from "../reducer";
+
+const Named = styled("span")`
+  font-weight: bold;
+`;
+
+interface Props {
+  datasetId: DatasetT["id"] | null;
+  row: EntityEvent;
+  column: ColumnDescription;
+  rootConceptId: ConceptIdT;
+}
+
+const ConceptName = ({ datasetId, row, column, rootConceptId }: Props) => {
+  // TODO: refactor. It's very implicit that the id is
+  // somehow containing the datasetId.
+  if (!datasetId) return null;
```

Comment:
where are you missing the datasetid? And where are you relying on it?

Reply:
The dataset id is available as soon as datasets have been loaded (it's not defined before that) – and now I need it here to create the full concept ID, I'd argue the backend should send the full ID as part of the csv, so including the dataset id